src/RevProc.cc

```diff
@@ -1564,6 +1566,31 @@ void RevProc::HandleALUFault(unsigned width){
                   "FAULT:ALU: ALU fault injected into next retire cycle\n");
 }
 
+bool RevProc::ThreadCanBeRemoved(const uint32_t& ThreadID) {
+  auto* regFile = AssignedThreads.at(ThreadID)->GetRegFile();
+
+  // Check the if regfile scoreboard is anything but clear 
+  // if not, this thread has outstanding requests and can't be removed
+  // TODO: (PRE-MERGE) Do this in a more ~Lee~ fashion
+  if( feature->IsRV32() ){
+    for( unsigned i=0; i<_REV_NUM_REGS_; i++ ){
+      if( regFile->RV_Scoreboard[i] ){
```

Comment:
Should be replaced with `regFile->RV_Scoreboard.any()`.

There's no need to differentiate with `feature->IsRV32()` anymore, since the scoreboards were merged.

Also, what about `regFile->FP_Scoreboard`?

Reply:
Whoops forgot to remove this function. I replaced it with `ThreadHasDependencies`